Update sdk to 14 and related fixes

davidyuk opened the pull request:
This PR is supported by the Æternity Foundation

https://github.com/aeternity/aepp-sdk-js/blob/develop/docs/guides/migration/14.md

found issues https://github.com/superhero-com/superhero-wallet/issues/3463 https://github.com/aeternity/aepp-sdk-js/issues/2046

davidyuk commented on src/styles/typography.scss:
```diff
@@ -1,3 +1,5 @@
+@use 'variables';
```
> Module Warning (from ./node_modules/sass-loader/dist/cjs.js):
Deprecation Warning on line 8, column 8 of file:///Users/denis/Documents/superhero-wallet/src/styles/typography.scss:8:8:
Sass @ import rules are deprecated and will be removed in Dart Sass 3.0.0.
More info and automated migrator: https://sass-lang.com/d/import

davidyuk commented on tests/e2e/integration/deeplinks.cy.js:
```diff
@@ -4,7 +4,7 @@ import { STUB_ACCOUNT, STUB_TX_BASE_64 } from '@/constants/stubs';
 const callbackUrl = encodeURIComponent('http://localhost');
 const encodedContractCallTx = encodeURIComponent('tx_+HQrAaEBbLQV9y00r/WCZ4XWrEVDpdiLPiLx2282rdKA1dkRTlmCA7yhBWuJbvnfcBIKpR/IbC+ywuM9P7fvmDFNerFKykCB+M1qA4al2v8FcAAAhzcV8VZnVACDGBf4hDuaygCQKxEV1igBG2+HI4byb8D/wPOLADU=');
-const encodedGaAttachTx = encodeURIComponent('tx_+O1QAaEByqPFadmQk4sGtyDiquosAZyKJNmherKOhheVIEYTLCITuIv4iUYDoIMndi6iAoqQltixF/3A6WlN0+rWVAan4LvFLy2pEJlhwLhcuDv+RNZEHwA3ADcAGg6CPwEDP/5s8lcLADcCFwcXdwAIPAIE+wNNTm90IGluIEF1dGggY29udGV4dAEBAJsvAhFE1kQfEWluaXQRbPJXCyVhdXRob3JpemWCLwCFNy4xLjAAoGzyVwsKFZm3CCkeUKo9rxPQx/JIS8M33a0kE6N/1KAJgwcAA4ZIjzqPaAAATIQ7msoAhysRRNZEHz+4yEN3');
+const encodedUnknownTx = encodeURIComponent('tx_+FKB/wGhAcqjxWnZkJOLBrcg4qrqLAGciiTZoXqyjoYXlSBGEywioQHKo8Vp2ZCTiwa3IOKq6iwBnIok2aF6so6GF5UgRhMsIguGDyb1YcgAABOAFw87cw==');
```
For this test better to choose a completely broken transaction, to ensure that the wallet is forward-compatible.

The problem I had with `GaAttachTx` is that currently, sdk rejects building such transactions with non-1 nonce.

> nonce should be equal 1 if GaAttachTx, got 19 instead

currently it will fail only if I request to replace the signer

davidyuk commented on src/protocols/aeternity/libs/AeternityAdapter.ts:
```diff
@@ -2,15 +2,16 @@
+  MemoryAccount,
   Tag,
 } from '@aeternity/aepp-sdk';
+import { getHdWalletAccountFromSeed } from '@aeternity/aepp-sdk-13';
```
I will make a new release of sdk to fix that. Meanwhile, you can use it as is since tree shaking will remove the unused code.

CedrikNikita commented on tests/e2e/support/coingecko-stubs.js:
```diff
@@ -0,0 +1,203 @@
+beforeEach(() => {
```
It was done here https://github.com/superhero-com/superhero-wallet/commit/aa852fa649a0a13deffb08038677492b413f8ddf .
Isn't it working properly?

CedrikNikita commented on src/content-scripts/inject.ts:
```diff
@@ -100,8 +101,7 @@ const runContentScript = () => {
       const port = browser.runtime.connect();
-      // @ts-expect-error
-      const extConnection = new BrowserRuntimeConnection({ port });
+      const extConnection = new BrowserRuntimeConnection({ port, debug: false });
```
Can you explain this change? What is pros/cons of this?

davidyuk replied:
This is actually a bug on sdk side that I'll fix in 12.1.0: `debug` shouldn't be required. From your side `@ts-expect-error` is an overkill because it can ignore a more significant TS error, its usage should be limited.

dvdbak commented:
Build URL is open in a black screen. Not able to test.

dvdbak commented:
It is not possible to import an AE account via private key.
Tryied with: 2602a31e7a3ce342a6f287ee10164cadb88fd16eaa883bd9b93a75f653be03322594d09b5907f8a6f1a9fd2c7e5b067720fe78d862e83b9d6a7ef0f0fb3173f2
![image](https://github.com/user-attachments/assets/c5657580-1bc4-4e56-b694-b5e5a005f60c)

dvdbak commented:
Unable to create a Gift card. created gift card has 0 value. 
on Mainnet and Testnet

https://github.com/user-attachments/assets/d20a2ef7-d21b-4efb-b914-d9ba107065a0

davidyuk commented:
I fixed the above two bugs, also
- added ability to add account by sk_-encoded private key, like sk_HjvV4utY9bQGZtp4vJ4mexCUjAoBcSW1mMT9Y9q4S6UYSqiqB
- improved `useStorageRef` types to fix warning
```
Invalid prop: type check failed for prop "secretKey". Expected Buffer, got Object  at <InviteItem>
at <IonContent>
at <IonPage>
at <Invite>
```

CedrikNikita commented on src/composables/storageRef.ts:
```diff
@@ -56,28 +59,28 @@ export function useStorageRef<T = string | object | any[]>(
-      restoredValue = await asyncPipe<T | null>(migrations)(restoredValue!);
+      restoredValue = await asyncPipe<TSerialized | null>(migrations)(restoredValue!);
       if (restoredValue !== null) {
-        await setStorageState(restoredValue);
```
I think `setStorageState` here was used in case we would want to introduce serializer to an existing (stored) value,

dvdbak commented:
#3498 Able to import twice the same account via sk_ prefixed key
@CedrikNikita Will it be handle by the Wallet?

davidyuk commented:
I've reverted the ability to import sk_-encoded private key, will do it separately
